app/components/goal-block/component.js

```diff
@@ -40,17 +44,23 @@ export default Component.extend(SwipableListItemMixin, {
     this.set('isActive', false);
   },
 
-  observePanOpen: observer('isPanOpen', function() {
-    if (this.get('isPanOpen')) {
+  setOptionWidth: observer('isEditing', function() {
+    if (this.get('isEditing')) {
       this.setProperties({
-        isEditing: false,
-        isEditingDisabled: true
+        optionWidth: SingleOptionWidth,
+        hasOneOption: true,
+        hasThreeOptions: false
       });
     } else {
       this.setProperties({
-        isEditingDisabled: false
+        optionWidth: SingleOptionWidth * 3,
+        hasOneOption: false,
+        hasThreeOptions: true
       });
```

Comment:
I'd get rid of hasThreeOptions and hasOne option and just have an optionCount, like we do in the main app. It's more re-usable and removes unnecessary complexity IMO.